test/WebJobs.Script.Tests.Integration/Security/AuthenticationLevelHandlerTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.WebHost;
+using Microsoft.Azure.WebJobs.Script.WebHost.Authentication;
+using Microsoft.Azure.WebJobs.Script.Workers.Rpc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.Json;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Hosting;
+using Microsoft.WebJobs.Script.Tests;
+using Moq;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.Integration.Security
+{
+    [Trait(TestTraits.Category, TestTraits.EndToEnd)]
+    [Trait(TestTraits.Group, nameof(AuthenticationLevelHandlerTests))]
+    public class AuthenticationLevelHandlerTests
+    // TODO: This should be a more general E2E test suite - maybe for SWA? Or combine with storage/secret E3E tests?
+    {
+        public static IEnumerable<object[]> GetData()
+        {
+            yield return new object[] { @"TestScripts\CSharp", RpcWorkerConstants.DotNetLanguageWorkerName,
+                new Dictionary<string, string>() { ["AzureWebJobsStorage"] = string.Empty }};
+
+            // test when AzureWebJobsSecretStorageType=None
+            // test when AzureWebJobsSecretStorageType= something else
+            // test when AzureWebJobsSecretStorageType is not set
+
+            // test for Node scenarios too
+        }
+
+        [Theory]
+        [MemberData(nameof(GetData))]
+        public async Task NoStorage_ReturnsExpected(string scriptRoot, string runtime, IDictionary<string, string> config)
+        {
+            // this initialization fails because ensuring the host has started up requires calling an admin endpoint
+            // need to update the TestFunctionHost.GetHostStatusAsync() method to use JWT bearer auth if secrets not enabled
+            using (var fixture = new AuthenticationLevelHandlerTestFixture(scriptRoot, runtime, config))
+            {
+                // call here to assert there's no connection string?
+
+                var code = "Puffin";
+
+                var response = await fixture.Host.HttpClient.GetAsync($"api/HttpTrigger?code={code}");
+                response.EnsureSuccessStatusCode();
+            }
+        }
+    }
+
+    public class AuthenticationLevelHandlerTestFixture : IDisposable
+    {
+        public TestFunctionHost Host;
+        private bool _disposed;
+
+        public AuthenticationLevelHandlerTestFixture(string scriptRoot, string testRuntime, IDictionary<string, string> webHostConfig)
```

Comment:
the webHostConfig you're passing in doesn't appear to be used? So how are you configuring the test host with SWA config?

Reply:
Thanks for the catch; I ended up not needing it. My understanding based on discussions with Fabio and Anthony is that we're this behavior change is meant to affect apps that meet the following two conditions, of which SWA apps are a subset:

1. No `AzureWebJobsStorage` setting / `AzureWebJobsStorage` is null/empty
2. `AzureWebEncryptionKey` exists (for JWT auth path)

Initially I set up webHostConfig to overwrite the test `AzureWebJobsStorage` setting with an empty value. Turns out if you do that, `IConfigurationBuilder` will go back through its sources for the last non-null/empty value for a setting, which is why `TestConfigurationBuilderExtensions.RemoveTestSettings()` now exists.

And 2 is set as an environment variable, so `webHostConfig` became an artifact. I've removed it.